src/sqlfluff/rules/L003.py

```diff
@@ -344,7 +344,7 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             return LintResult(memory=memory)
 
         if raw_stack and raw_stack[-1] is not context.segment:
-            raw_stack = raw_stack + (context.segment,)
+            raw_stack = raw_stack + (context.segment,)  # type: ignore
```

Comment:
Think it would be worth making a proper fix rather than `type: ignore`? I did some tinkering, and we can fix it properly by changing 5 lines. (Happy to share that fix if you like.)

Reply:
Sounds good, you can just make a new pr is easier

Reply:
Will do!

Reply:
Created a new PR. mypy seemed happy locally. Let's see if the build is happy. #2127